### PR TITLE
Make CRIB config more modular, decouple devspace from helm charts

### PR DESCRIPTION
**Pull request by scheibinger**
## Motivation
* Relying on sync between core chart and ccip chart is painful and slow down crib development. 
* ccip-scripts and other ccip related features were mixed up with chaninlink cluster and that make it harder to configure and manage
* With new use cases, crib config increases in complexity. We should have a way to easily add new features, ideally using subcharts, for config isolation

## Solution
Split existing chanlink-cluster chart into root and subcharts:
* root chart will be directly referenced by devspace and argocd to allow single command deployment
* chainlink-cluster chart now it is inline more less with chainlink-cluster chart in the core repo. I will replace the reference in the future PR
* ccip-scripts is additional feature, only required for ccip, so it make sense to keep it under separate chart

The new directory structure for Crib would be following:
```
charts/
   root/ (main chart used by devspace and argocd)
     values.yaml
   feature-a/
   feature-b/
crib/
   devspace.yaml
   
```

## Testing
- [x] CLI Flow
- [x] PR Flow

### CODE REVIEW

File: charts/root/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# It is recommended to use it with quotes.
+appVersion: "0.1.0"
+dependencies:
+  - name: chainlink-cluster
+    version: 0.4.1
+    repository: "file://../chainlink-cluster"
+  - name: ccip-scripts
+    version: 0.1.0
+    repository: "file://../ccip-scripts"
+#    condition: ccip.deployContractsAndJobs
```
Comment (chainchad) [inline charts/root/Chart.yaml]: Curious as to why this is commented?

File: charts/root/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "0.1.0"
+dependencies:
+  - name: chainlink-cluster
+    version: 0.4.1
+    repository: "file://../chainlink-cluster"
```
Comment (chainchad) [inline charts/root/Chart.yaml:28]: Is this valid? I don't see a Chart.yaml inside here.

Comment (scheibinger) [reply charts/root/Chart.yaml:28]: it worked, it is relative to the current chart directory, I took it from the example here: https://helm.sh/docs/helm/helm_dependency/#synopsis

It is also temporary until I fix the helm release in core here: https://github.com/smartcontractkit/chainlink/pull/12858

File: charts/root/Chart.yaml (hunk 1 above)
Comment (scheibinger) [reply charts/root/Chart.yaml]: ah, I missed that. Using such conditions is a nice way to disable feature in Crib setup

Comment (scheibinger) [reply charts/root/Chart.yaml]: fixed

File: charts/root/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "0.1.0"
+dependencies:
+  - name: chainlink-cluster
```
Comment (scheibinger) [inline charts/root/Chart.yaml:26]: change all the names to camelcase

File: charts/root/Chart.lock [hunk 4]
```diff
@@ -0,0 +1,9 @@
+dependencies:
+- name: chainlink-cluster
```
Comment (chainchad) [inline charts/root/Chart.lock:2]: TODO: change names to camel case

File: crib/README.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
+# Crib Devspace Setup
+
+Crib is a devspace configuration to launch CCIP enabled chainlink cluster
```
Comment (chainchad) [inline crib/README.md]:
```suggestion
CRIB is a devspace configuration to launch CCIP enabled chainlink cluster 
```

**Review by chainchad — CHANGES_REQUESTED** [review]
LGTM once we update names to camelCase and change name to "CRIB". Do we need to update any docs?

**Review by chainchad — APPROVED** [review]

**Review by skudasov — APPROVED** [review]

**Review by chainchad — APPROVED** [review]

**Pull request merged**